plugins/speculation-rules/load.php

```diff
@@ -20,15 +20,69 @@
 	exit;
 }
 
-// Define the constant.
-if ( defined( 'SPECULATION_RULES_VERSION' ) ) {
-	return;
-}
+(
+	/**
+	 * Register this copy of the plugin among other potential copies.
+	 *
+	 * @param string  $global_var_name Global variable name for storing the plugin pending loading.
+	 * @param string  $version         Version.
+	 * @param Closure $load            Callback that loads the plugin.
+	 */
+	static function ( string $global_var_name, string $version, Closure $load ) {
+		if ( ! isset( $GLOBALS[ $global_var_name ] ) ) {
+			$bootstrap = static function () use ( $global_var_name ) {
+				if (
+					isset( $GLOBALS[ $global_var_name ]['load'], $GLOBALS[ $global_var_name ]['version'] )
+					&&
+					$GLOBALS[ $global_var_name ]['load'] instanceof Closure
+					&&
+					is_string( $GLOBALS[ $global_var_name ]['version'] )
+				) {
+					call_user_func( $GLOBALS[ $global_var_name ]['load'], $GLOBALS[ $global_var_name ]['version'] );
+					unset( $GLOBALS[ $global_var_name ] );
+				}
+			};
+
+			// Handle either where the plugin is installed as a regular plugin or is embedded in another plugin or in a theme.
+			if ( ! did_action( 'plugins_loaded' ) ) {
+				add_action( 'plugins_loaded', $bootstrap, 0 );
+			}
+
+			// Handle case where plugin is embedded in a theme.
+			add_action( 'after_setup_theme', $bootstrap, 0 );
+		}
+
+		// Register this copy of the plugin.
+		if (
+			// Register this copy if none has been registered yet.
+			! isset( $GLOBALS[ $global_var_name ]['version'] )
+			||
+			// Or register this copy if the version greater than what is currently registered.
+			version_compare( $version, $GLOBALS[ $global_var_name ]['version'], '>' )
+			||
+			// Otherwise, register this copy if it is actually the one installed in the directory for plugins.
+			rtrim( WP_PLUGIN_DIR, '/' ) === dirname( __DIR__ )
+		) {
+			$GLOBALS[ $global_var_name ]['version'] = $version;
+			$GLOBALS[ $global_var_name ]['load']    = $load;
+		}
+	}
+)(
+	'plsr_pending_plugin_info',
+	'1.2.1',
+	static function ( string $version ) {
```

Comment:
The version needn't be passed to the load callback as explained in https://github.com/WordPress/performance/pull/1159/files#r1571042878. I did this just to reduce duplication of the version in this file, but if it is already present twice it doesn't hurt to have it present three times.

```suggestion
					isset( $GLOBALS[ $global_var_name ]['load'] )
					&&
					$GLOBALS[ $global_var_name ]['load'] instanceof Closure
				) {
					call_user_func( $GLOBALS[ $global_var_name ]['load'] );
					unset( $GLOBALS[ $global_var_name ] );
				}
			};

			// Handle either where the plugin is installed as a regular plugin or is embedded in another plugin or in a theme.
			if ( ! did_action( 'plugins_loaded' ) ) {
				add_action( 'plugins_loaded', $bootstrap, 0 );
			}

			// Handle case where plugin is embedded in a theme.
			add_action( 'after_setup_theme', $bootstrap, 0 );
		}

		// Register this copy of the plugin.
		if (
			// Register this copy if none has been registered yet.
			! isset( $GLOBALS[ $global_var_name ]['version'] )
			||
			// Or register this copy if the version greater than what is currently registered.
			version_compare( $version, $GLOBALS[ $global_var_name ]['version'], '>' )
			||
			// Otherwise, register this copy if it is actually the one installed in the directory for plugins.
			rtrim( WP_PLUGIN_DIR, '/' ) === dirname( __DIR__ )
		) {
			$GLOBALS[ $global_var_name ]['version'] = $version;
			$GLOBALS[ $global_var_name ]['load']    = $load;
		}
	}
)(
	'plsr_pending_plugin_info',
	'1.2.1',
	static function () {
```

Reply:
Eesh. This suggestion diff is not helpful to read! (It just removes `$version` from the `$load` callback.)